src/Controllers/controller.py

```diff
@@ -37,8 +38,9 @@ def run(self):
                 self.quit()
             except EOFError:
                 self.quit()
-            except:
-                self._states.undo(self._diagram)   
+            except Exception as e:
+                print(str(e))
+                self._states.undo()   
```

Comment:
If you haven't, well want to make sure we test this thoroughly to make sure it doesn't undo something that was valid.